harpy/scripts/demultiplex_gen1.py

```
@@ -111,8 +110,8 @@ def get_read_codes(index_read, left_segment, right_segment):
 R1_output = {}
 R2_output = {}
 for sample in samples:
-    R1_output[sample] = subprocess.Popen(["gzip"], stdin = subprocess.PIPE , stdout = open(f"{outdir}/{sample}.{part}.R1.fq.gz", 'wb'))
-    R2_output[sample] = subprocess.Popen(["gzip"], stdin = subprocess.PIPE , stdout = open(f"{outdir}/{sample}.{part}.R2.fq.gz", 'wb'))
+    R1_output[sample] = open(f"{outdir}/{sample}.{part}.R1.fq", 'w')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)